ARGO/src/AssetManager.cpp

```diff
@@ -0,0 +1,237 @@
+#include "stdafx.h"
+#include "AssetManager.h"
+
+AssetManager* AssetManager::s_instance = NULL;
+
+AssetManager* AssetManager::Instance(SDL_Renderer& t_renderer)
+{
+	if (s_instance == NULL)
+	{
+		s_instance = new AssetManager(t_renderer);
+	}
+	return s_instance;
+}
+
+AssetManager* AssetManager::Instance()
+{
+	return s_instance;
+}
+
+void AssetManager::Release()
+{
+	delete s_instance;
+	s_instance = NULL;
+}
+
+SDL_Texture* AssetManager::GetTexture(std::string t_filename)
+{
+	std::string fullPath = SDL_GetBasePath();
+	fullPath.append(Utilities::IMAGES_PATH + t_filename);
+
+	if (!m_textures[fullPath])
+	{
+		m_textures.at(fullPath) = loadTexture(fullPath);
```

Comment:
why using .at() on this line but using [] elsewhere in the function?

Reply:
same for get sfx function and getmusic function

Reply:
map.at(key) throws exception when missing a key while map[key] enters a key if it doesn't exist in the map you plebian
http://www.cplusplus.com/reference/map/map/operator[]/
@CathalRedmond